shared/logger.go

```diff
@@ -54,45 +54,56 @@ func NewNilLogger() Logger {
 	return nl
 }
 
-type gaeLogger struct {
-	ctx context.Context
-}
-
-func (l gaeLogger) Debugf(format string, args ...interface{}) {
-	gaelog.Debugf(l.ctx, format, args...)
-}
-
-func (l gaeLogger) Errorf(format string, args ...interface{}) {
-	gaelog.Errorf(l.ctx, format, args...)
-}
+// HandleWithStandardGCL - TODO: Refactor
+func HandleWithStandardGCL(h http.HandlerFunc, project string, childLogger, parentLogger *gclog.Logger) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if isDevAppserver() || childLogger == nil || parentLogger == nil {
+			h(w, r.WithContext(withLogger(r.Context(), logrus.New())))
+			return
+		}
 
-func (l gaeLogger) Infof(format string, args ...interface{}) {
-	gaelog.Infof(l.ctx, format, args...)
-}
+		start := time.Now()
+		// See https://cloud.google.com/appengine/docs/flexible/go/writing-application-logs
+		traceID := strings.Split(r.Header.Get("X-Cloud-Trace-Context"), "/")[0]
+		if traceID != "" {
+			traceID = fmt.Sprintf("projects/%s/traces/%s", project, traceID)
+		}
 
-func (l gaeLogger) Warningf(format string, args ...interface{}) {
-	gaelog.Warningf(l.ctx, format, args...)
-}
+		gcl := gcLogger{
+			childLogger: childLogger,
+			traceID:     traceID,
+			maxSeverity: gclog.Default,
```

Comment:
maybe change the default severity to Info, consistent with the appengine default level